src/check.rs

```diff
@@ -0,0 +1,88 @@
+use core::str;
+
+use esp_metadata::Chip;
+
+struct Version {
+    major: u8,
+    minor: u8,
+    patch: u8,
+}
+
+enum CheckResult {
+    Ok,
+    WrongVersion,
+    NotFound,
+}
+
+pub fn check(chip: Chip) {
+    let rust_version = get_version(
+        "cargo",
+        if chip.is_xtensa() {
+            &["+esp"]
+        } else {
+            &["+stable"]
+        },
+    );
+
+    let espflash_version = get_version("espflash", &[]);
+    let probers_version = get_version("probe-rs", &[]);
+
+    println!("\nChecking installed versions");
+    print_result("Rust", check_version(rust_version, 1, 84, 0));
```

Comment:
How do we make sure we keep these up to date?

Reply:
We could query gh for the latest version. We could copy most of the code from`espup`.

Reply:
I creates #89 to not forget about this